### ISSUE TITLE
Compatibility models keras 1.0.4 and 1.1.0

### ISSUE THREAD

**Issue by acasamitjana** [fchollet/keras#3957]
Has anyone faced problems when loading weights from a model trained using a different keras versions?

I trained my model using Keras 1.0.4. Then after upgrading to 1.1.0, I tried to load the weights to the same model but it doesn't fit it.

Error:
Exception: Layer #23 (named "upper_conv_4" in the current model) was found to correspond to layer batchnormalization_16 in the save file. However the new layr upper_conv_4 expects 2 weights, but the saved weights have 4 elements.

Both layers exist in the model but it seems to create layers in a different way, since in "model.summary()" the appear in a different order.

**Comment by keunwoochoi** [comment]
Probably you should re-train the model. The new BN (perhaps since keras>=1.0.6) is not compatible to the previous ones.

**Comment by ksze** [comment]
There is no way to just upgrade the model without completely retraining it?
